### PR TITLE
Add Documentation for Adding Two Numbers Represented as Linked Lists

### LINKED ISSUE #1064: [Idea]: Add Documentation for "Add Two Numbers Represented as Linked Lists" [LEVEL-Medium]
### Idea Title

Add Two Numbers Represented as Linked Lists.

### Idea Description

I would like to propose adding a documenation to the folder "DSA-Problem-Solution" for the "Add Two Numbers Represented as Linked Lists" functionality. This documentation will enhance user understanding and provide clear guidance on how to solve this.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by riyaa060**
### Description
This PR adds comprehensive documentation explaining the solution to the problem of adding two numbers represented as linked lists. 
Fixes #1064

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/DSA-Problem-Solution/Add_two_number_as_LL.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
+---
+id: add-two-numbers-as-linked-lists
+title: "Add two numbers represented as linked lists"
+sidebar_label:  "Adding Two Numbers Represented as Linked Lists"
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Add_two_number_as_LL.md]: write in sort word for our sidebar

File: docs/DSA-Problem-Solution/Delete_occurences_of_key.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
+---
+id: delete-all-occurences-of-key
+title: "Deleting All Occurrences of a Key in a Doubly Linked List"
+sidebar_label: "Deleting All Occurrences of a Key in a Doubly Linked List
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Delete_occurences_of_key.md]: write in sort word for our sidebar

File: docs/DSA-Problem-Solution/Delete_occurences_of_key.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+sidebar_position: 12
+description: "This document provides a detailed explanation and implementation for deleting all occurrences of a key in a doubly linked list , including step-by-step instructions and example code."
+tags: [Linked list,Deletion]
+---
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Delete_occurences_of_key.md]: can not read a block mapping entry; a multiline key may not be an implicit key.
like:
`name: null,`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by riyaa060** [comment]
Check @ajay-dhangar

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**